segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -167,6 +172,56 @@ public void setupAppend(SetupAppend setupAppend) {
                 });
     }
 
+    @VisibleForTesting
+    CompletableFuture<Void> setupTokenExpiryTask(SetupAppend setupAppend) {
+        String segment = setupAppend.getSegment();
+        UUID writerId = setupAppend.getWriterId();
+        long requestId = setupAppend.getRequestId();
+
+        Duration expiryDuration = this.durationToExpiry(setupAppend);
+
+        if (expiryDuration == null) {
+            return CompletableFuture.completedFuture(null);
+        } else {
+            return Futures.delayedTask(() -> {
+                if (isSetupAppendCompleted(segment, writerId)) {
+                    try {
+                        log.trace("Informing writer {} that sent request {}, about token expiry for segment {}",
+                                writerId, requestId, segment);
+                        if (!connection.isClosed()) {
```

Comment:
we do not need this check since we are closing the connection.

Reply:
IMO, this check is still useful because this task is a background task and the connection might already have closed out-of-band.

Reply:
I've moved this check inside close itself, and have removed it here.